src/map/pc.cpp

```diff
@@ -2427,14 +2427,16 @@ void pc_clean_skilltree(struct map_session_data *sd)
 uint64 pc_calc_skilltree_normalize_job_sub( struct map_session_data *sd ){
 	int skill_point = pc_calc_skillpoint( sd );
 
-	if( sd->class_ & MAPID_SUMMONER ){
-		// Novice's skill points for basic skill.
+	if( sd->class_ & MAPID_SUMMONER || sd->class_ & MAPID_SPIRIT_HANDLER ){
+		// Summoner's skill points for base skills.
 		std::shared_ptr<s_job_info> summoner_job = job_db.find( JOB_SUMMONER );
 
 		int summoner_skills = summoner_job->max_job_level - 1;
 
 		if( skill_point < summoner_skills ){
 			return MAPID_SUMMONER;
+		} else {
+			return MAPID_SPIRIT_HANDLER;
 		}
```

Comment:
Through this change all adjustments below are useless.
I did not add it like this to make it more dynamic.
But seems like I messed up somewhere and thats the real issue we need to look into.
I think it is because Spirit Handler has JOBL_2_1, but would need to check it in more detail.